blob/blob.go

```diff
@@ -12,8 +12,26 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-// Package blob provides an easy way to interact with Blob objects within
-// a bucket. It utilizes standard io packages to handle reads and writes.
+// Package blob provides an easy and portable way to interact with blob objects
+// within a container, hereafter called the "bucket".
+//
+// It supports operations like reading and writing blobs (using standard
+// interfaces from the io package), deleting a blob, and listing blobs in a
+// bucket.
+//
+// To construct a *Bucket, don't use the NewBucket function in this package.
+// Instead, import one of the blob subpackages (e.g., fileblob, gcsblob,
+// s3blob), and use its exported function(s).
```

Comment:
It would be good to add an example here, like:

```suggestion
// s3blob), and use its exported function(s). For example:
//
//	bucket, err := fileblob.OpenBucket("path/to/dir", nil)
//	if err != nil {
//		return fmt.Errorf("could not open bucket: %v", err)
//	}
```

Reply:
OK. There's also an actual example, but inline is fine too.